### ISSUE TITLE
--retry-count flag should have an effect

### ISSUE THREAD

**Issue by esemeniuc** [peak/s5cmd#234]
Hi, when running `s5cmd` with the retry flag, I get errors like this:

```
$ s5cmd --retry-count 20000 cp -c 1 "backup/*" s3://abc/eric/
ERROR "cp backup/foo.crc s3://abc/eric/foo.crc": RequestError: send request failed caused by: Put "https://bucket.s3.us-west-2.amazonaws.com/eric/foo.crc": read tcp 10.55.10.38:62285->52.218.235.81:443: read: connection reset by peer
```
I believe 20000 retries should catch all errors. 

Looking at the code at `s3.go`, I see that `ShouldRetry` is never called. Any thoughts?
https://github.com/peak/s5cmd/blob/910cf214d881dca000c2f9d7eefcc332f5150247/storage/s3.go#L700-L710

**Comment by igungor** [comment]
Hi @esemeniuc 

What's your s5cmd version?

**Comment by esemeniuc** [comment]
Hi @igungor 

Here is the version:
```
$ ./s5cmd version
v1.1.0-4c36b52
```

**Comment by igungor** [comment]
> I believe 20000 retries should catch all errors.

`--retry-count` is only effective for retryable errors. Please see https://github.com/peak/s5cmd#retry-logic

> Looking at the code at s3.go, I see that ShouldRetry is never called. Any thoughts?

`connection reset by peer` errors are retried on `master` branch. It's just not released yet.

Thanks for the report.

**Comment by igungor** [comment]
Fixed with release v1.2.0.
